In [8]:
import numpy as np
import pandas as pd

In [9]:
data = pd.read_excel('Business Analyst -ABC University.xlsx')

In [12]:
data_size = data.size
print(f"The dataset size is: {data_size}")

The dataset size is: 695288


In [17]:
rows, columns = data.shape
print(f"The dataset has {rows} rows and {columns} columns.")

The dataset has 31604 rows and 22 columns.


In [15]:
data.head(10)

Name      Mobile Lead Origin Country        State     City  \
0          Ram Yadav  9830******         API   India  West Bengal  Kolkata   
1    Anamika Bairagi  7439******         API   India  West Bengal  Barasat   
2         Puja Singh  7003******         API   India  West Bengal  Kolkata   
3       Jhuma Haldar  9831******         API   India  West Bengal  Kolkata   
4        Rajan Kumar  9128******         API   India  West Bengal  Kolkata   
5   Deepshikha kamti  6291******         API   India  West Bengal  Kolkata   
6       Shivang soni  9674******         API   India  West Bengal  Kolkata   
7        Kajol Rathi  9748******         API   India  West Bengal  Kolkata   
8  Sayon chakraborty  9804******         API   India  West Bengal  Kolkata   
9   Sk Zahira Khatun  9831******         API   India  West Bengal  Kolkata   

  Instance Instance Date        Campaign    Lead Stage  ...  \
0  Primary    2023-10-09     Meta/85/API     Untouched  ...   
1  Primary    2023-10-09     Meta/84/API  Disqualified  ...   
2  Primary    2023-10-09     Meta/85/API   No Response  ...   
3  Primary    2023-10-09     Meta/88/API  Disqualified  ...   
4  Primary    2023-10-09     Meta/85/API  Disqualified  ...   
5  Primary    2023-10-09     Meta/85/API     Untouched  ...   
6  Primary    2023-10-09     Meta/85/API  Disqualified  ...   
7  Primary    2023-10-09     Meta/88/API  Disqualified  ...   
8  Primary    2023-10-09  Meta/90_RV/API   No Response  ...   
9  Primary    2023-10-09     Meta/88/API  Disqualified  ...   

  Mobile Verification Status Registration Device  \
0                         No                 NaN   
1                         No                 NaN   
2                         No                 NaN   
3                         No                 NaN   
4                         No                 NaN   
5                         No                 NaN   
6                         No                 NaN   
7                         No                 NaN   
8                         No                 NaN   
9                         No                 NaN   

                                              Course  \
0                   B.Tech in Biomedical Engineering   
1  BCA (Hons) in Banking, Financial Services and ...   
2           MBA (Human Resource, Marketing, Finance)   
3                               M.A in Public Policy   
4                                          B.Com (H)   
5                                             M.Com.   
6       MBA in Logistics and Supply Chain Management   
7                                             M.Com.   
8  BCA (Hons) in Banking, Financial Services and ...   
9                          B.A (Hons.) Public Policy   

                 Specialization Campus Last Lead Activity Date Form Initiated  \
0  Specialization Not Available     UG     2023-10-09 11:26:00              0   
1  Specialization Not Available     UG     2023-10-09 11:20:00              0   
2  Specialization Not Available     PG     2023-10-09 10:17:00              0   
3  Specialization Not Available     PG     2023-10-09 10:16:00              0   
4  Specialization Not Available     UG     2023-10-09 08:14:00              0   
5  Specialization Not Available     PG     2023-10-09 07:05:00              0   
6  Specialization Not Available     PG     2023-10-09 05:26:00              0   
7  Specialization Not Available     PG     2023-10-09 03:39:00              0   
8  Specialization Not Available     UG     2023-10-09 02:35:00              0   
9  Specialization Not Available     UG     2023-10-09 01:40:00              0   

  Paid Applications  Submitted Applications  Enrolment Status  
0                 0                       0                No  
1                 0                       0                No  
2                 0                       0                No  
3                 0                       0                No  
4                 0                       0                No

#### Data Cleaning for more accurate results

In [21]:
missing_values = data.isnull().sum()
print(missing_values)

Name                              0
Mobile                            0
Lead Origin                       0
Country                           0
State                             0
City                              0
Instance                          0
Instance Date                     0
Campaign                          0
Lead Stage                        0
Lead Status                       0
Email Verification Status         0
Mobile Verification Status        0
Registration Device           30912
Course                            0
Specialization                    0
Campus                            0
Last Lead Activity Date           0
Form Initiated                    0
Paid Applications                 0
Submitted Applications            0
Enrolment Status                  0
dtype: int64


##### After deleting the 30912 missing values we only left with the data having registration devices

In [26]:
regDevice_data = data.dropna()
regDevice_data.head()

Name      Mobile Lead Origin Country        State  \
279       Sneha Adhikary  6290******      Online   India  West Bengal   
344        Rahima khatun  7318******      Online   India  West Bengal   
1810      ABHISHEK KUMAR  7352******      Online   India        Bihar   
1826  Kankan Chakraborty  7584******      Online   India  West Bengal   
1830      Srishti Kumari  9060******      Online   India        Bihar   

           City Instance Instance Date               Campaign    Lead Stage  \
279     Kolkata  Primary    2023-10-04  Meta/aTextlink/Online     Qualified   
344    Jangipur  Primary    2023-10-03    Meta/Emailer/Online     Untouched   
1810      Patna  Primary    2023-09-12    Meta/Emailer/Online  Disqualified   
1826    Kolkata  Primary    2023-09-12    Meta/Emailer/Online  Disqualified   
1830  Madhepura  Primary    2023-09-12    Meta/Emailer/Online  Disqualified   

      ... Mobile Verification Status Registration Device  \
279   ...                        Yes             Desktop   
344   ...                        Yes              Mobile   
1810  ...                        Yes              Mobile   
1826  ...                        Yes              Mobile   
1830  ...                        Yes              Mobile   

                                        Course  \
279              M.Sc in Environmental Science   
344                                    Bengali   
1810  M.Sc. Tech (Statistics and Data Science)   
1826                         M.Sc in Chemistry   
1830                         M.Sc in Chemistry   

                                         Specialization Campus  \
279        School of Basic and Applied Sciences (SOBAS)     PG   
344   School of Liberal Arts and Culture Studies (SO...    PhD   
1810       School of Basic and Applied Sciences (SOBAS)     PG   
1826       School of Basic and Applied Sciences (SOBAS)     PG   
1830       School of Basic and Applied Sciences (SOBAS)     PG   

     Last Lead Activity Date Form Initiated Paid Applications  \
279      2023-10-04 22:00:00              1                 0   
344      2023-10-03 18:44:00              1                 0   
1810     2023-09-26 11:01:00              0                 0   
1826     2023-09-26 11:01:00              0                 0   
1830     2023-09-26 11:01:00              1                 0   

      Submitted Applications  Enrolment Status  
279                        0                No  
344                        0                No  
1810                       0                No  
1826                       0                No  
1830                       0                No  

[5 rows x 22 columns]

##### Find Duplicates

In [37]:
duplicates = data.duplicated().sum()
print(f"There are {duplicates} Duplicate Values in the Dataset")

There are 0 Duplicate Values in the Dataset
